js/apps/admin-ui/src/realm-settings/user-profile/AttributesGroupForm.tsx

```diff
@@ -110,82 +110,63 @@ export default function AttributesGroupForm() {
         titleKey={matchingGroup ? "editGroupText" : "createGroupText"}
         divider
       />
-      <PageSection variant="light" onSubmit={form.handleSubmit(onSubmit)}>
-        <FormAccess isHorizontal role="manage-realm">
-          <FormGroup
-            label={t("nameField")}
-            fieldId="kc-name"
-            isRequired
-            helperTextInvalid={t("required")}
-            validated={form.formState.errors.name ? "error" : "default"}
-            labelIcon={
-              <HelpItem helpText={t("nameHintHelp")} fieldLabelId="nameField" />
-            }
+      <PageSection variant="light">
+        <FormProvider {...form}>
+          <FormAccess
+            isHorizontal
+            role="manage-realm"
+            onSubmit={handleSubmit(onSubmit)}
           >
-            <KeycloakTextInput
-              id="kc-name"
-              isReadOnly={!!matchingGroup}
-              {...form.register("name", { required: true })}
+            <TextControl
+              name="name"
+              label={t("nameField")}
+              labelIcon={t("nameHintHelp")}
+              rules={{ required: t("required") }}
+              readOnlyVariant={matchingGroup ? "default" : undefined}
```

Comment:
Let's use `readOnly` here, we don't care too much about the variant, and it's closer to the prop a native `input` would use, as well as a bit more succinct.
```suggestion
              readOnly={!!matchingGroup}
```

Reply:
ok thanks, fixed